lib/improver/tests/ensemble_calibration/ensemble_calibration_utilities/test_SplitHistoricForecastAndTruth.py

```diff
@@ -0,0 +1,223 @@
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""
+Unit tests for the `SplitHistoricForecastAndTruth` plugin.
+
+"""
+import datetime
+import unittest
+
+import iris
+from iris.tests import IrisTest
+import numpy as np
+
+from improver.ensemble_calibration.ensemble_calibration_utilities import (
+    SplitHistoricForecastAndTruth)
+from improver.tests.set_up_test_cubes import set_up_variable_cube
+
+
+class SetupCubesAndDicts(IrisTest):
+    """Set up historical forecast and truth cubes and associated dictionaries
+    for use in testing."""
+
+    def setUp(self):
+        """Set up dictionaries for testing."""
+        # Create historic forecasts and truth cubelists
+        data = np.array([[1., 2.],
+                         [3., 4.]], dtype=np.float32)
+        frt_dt = datetime.datetime(2017, 11, 10, 0, 0)
+        time_dt = datetime.datetime(2017, 11, 10, 4, 0)
+        self.historic_forecasts = iris.cube.CubeList([])
+        self.truth = iris.cube.CubeList([])
+        for day in range(6):
+            new_frt_dt = frt_dt + datetime.timedelta(days=day)
+            new_time_dt = time_dt + datetime.timedelta(days=day)
+            self.historic_forecasts.append(
+                set_up_variable_cube(data, time=new_time_dt, frt=new_frt_dt,
+                                     standard_grid_metadata="uk_ens"))
+            self.truth.append(
+                set_up_variable_cube(data, time=new_time_dt, frt=new_time_dt,
+                                     standard_grid_metadata="uk_det"))
+
+        self.combined = self.historic_forecasts + self.truth
+
+        # Create the historic and truth cubes
+        self.historic_forecasts_cube = self.historic_forecasts.merge_cube()
+        self.truth_cube = self.truth.merge_cube()
+
+        # Create historical forecasts and truth cubes and cubelists where
+        # some items are missing.
+        self.partial_historic_forecasts = (
+            self.historic_forecasts[:2] + self.historic_forecasts[3:])
+        self.partial_historic_forecasts_cube = (
+            self.partial_historic_forecasts.merge_cube())
+        self.partial_truth = self.truth[:2] + self.truth[3:]
+        self.partial_truth_cube = self.partial_truth.merge_cube()
+
+        # Set up dictionaries specifying the metadata to identify the
+        # historical forecasts and truth
+        self.historic_forecast_dict = {
+            "attributes": {
+                "mosg__model_configuration": "uk_ens"
```

Comment:
Just to let you know, it came up today that using mosg__ attributes aren't very future proof as they aren't very future proof. At some point we may need to add some kind of "truth_or_forecast" attribute to the cubes as they are stored in the EMOS cache and use this EMOS specific attribute.

Reply:
Thanks. For the moment, I'm using the attributes but this could be modified in the future, if the metadata conventions change.

Reply:
Caroline S was going to make a note of this in the relevant metadata epic. It might be that we need to update this to use be able to use a coordinate instead at some point.